src/main/java/org/xbill/DNS/ResolverConfig.java

```diff
@@ -355,71 +359,62 @@ else if(line.startsWith("options")) {
 				readingServers = true;
 			}
 		}
-		
+		if (lserver.isEmpty()) {
+			return false;
+		}
 		configureFromLists(lserver, lsearch);
+		return true;
 	}
 	catch (IOException e) {
+		return false;
 	}
 }
 
-private void
+private boolean
 findWin(InputStream in) {
+	boolean found;
 	String property = "org.xbill.DNS.windows.parse.buffer";
 	final int defaultBufSize = 8 * 1024;
 	int bufSize = Integer.getInteger(property, defaultBufSize);
 	BufferedInputStream b = new BufferedInputStream(in, bufSize);
 	b.mark(bufSize);
-	findWin(b, null);
+	found = findWin(b, null);
 	if (servers == null) {
 		try {
 			b.reset();
 		} 
 		catch (IOException e) {
-			return;
+			return false;
 		}
-		findWin(b, new Locale("", ""));
-	}
-}
-
-/**
- * Calls winipcfg and parses the result to find servers and a search path.
- */
-private void
-find95() {
-	String s = "winipcfg.out";
-	try {
-		Process p;
-		p = Runtime.getRuntime().exec("winipcfg /all /batch " + s);
-		p.waitFor();
-		File f = new File(s);
-		findWin(new FileInputStream(f));
-		new File(s).delete();
-	}
-	catch (Exception e) {
+		found = findWin(b, new Locale("", ""));
 	}
+	return found;
 }
 
 /**
  * Calls ipconfig and parses the result to find servers and a search path.
  */
-private void
-findNT() {
+boolean
+findWin() {
+	boolean found;
 	try {
 		Process p;
 		p = Runtime.getRuntime().exec("ipconfig /all");
-		findWin(p.getInputStream());
+		found = findWin(p.getInputStream());
```

Comment:
netsh is still locale-dependent (on my Windows 10 with Spanish set):
```
C:\...>netsh int ipv4 show dnsservers
Configuración para la interfaz "Ethernet"
    Servidores DNS configurados a través de DHCP:  192.168.1.1
    Registrar con el sufijo:           Solo el principal
...
```

That said, it's probably a better source to parse since it eliminates most of the noise

Reply:
It lacks the search domains though, not sure atm. if there's another netsh call for those.

Reply:
I still think we should consider the registry option.  What's wrong with looking at:
`regedit /e TEMPFILE "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip\Parameters"`
and parse those?

Reply:
You'll get nameservers of NICs with a disconnected link. Example: laptop with a wired and WiFi NIC. WiFi is connected at home to your router and the DHCP server assigns an IP and DNS addresses. Later you go to work, connect via cable (e.g. docking station), WiFi is disconnected. Your regedit call (and what Java does internally with probing the registry and calling Windows 95 (!) networking APIs) will get the inaccessible nameservers of your home WiFi.
Or the other way around (office nameservers still on the wired NIC, now only a WiFi connection). Querying the registry for this info is just _completely_ wrong. See also [JDK-7006496](https://bugs.openjdk.java.net/browse/JDK-7006496).

What I'd be fine with though is adding an _optional_ (i.e. scope=provided) dependency to JNA, probing at runtime if it's available and then doing proper calls to Iphlpapi. But that API isn't easy.

Reply:
name servers sure, but what about search suffixes?  Regardless of whether you disconnect and reconnect at different places, do the DNS Search Suffixes change?

Reply:
Yes, they do change, at least their priority. At work I have `ds.example.com`, at home my cable router hands out `home`. Fritz Boxes (popular routers/modems in Europe and especially Germany) assign `fritz.box`.

Reply:
Interesting.  Can we table this for a different PR/issue?  How we go about adding new functionality to better handle Windows resolution specifically could go on for awhile and we probably should discuss it in a separate issue.  I definitely don't want to tackle native code additions here.

Reply:
Sure, this was just about the question if netsh is also locale dependent. It is, so, bummer.

Reply:
Logged issue #55 to track the work to call Iphlpapi